tests/integration/serve_tests.rs

```diff
@@ -171,6 +171,43 @@ impl ServeClient {
 
     return self.endpoint.borrow().clone().unwrap();
   }
+
+  // Check if a new browser tab has been opened by checking
+  // stderr includes the target output expected
+  fn check_tab_opened(&self) {
```

Comment:
I'm not sure if this test makes sense - on CI there will be no browser to open, while locally people will suddenly get some new tabs. I'd suggest to not add it at all.